.github/workflows/docker.yml

```diff
@@ -0,0 +1,22 @@
+name: Docker Image
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and push
```

Comment:
I think it will upload a new main image even before the pull requests are accepted. Which may not be the desired action.

Reply:
It will push only if the `push` option is set to `true`, but I haven't done this yet.

This is still a very early WIP.

Reply:
Ah, ok. In that case sorry for the random/early comment ;-) 

Meantime I found that:

```
if: github.event_name == 'push'
```

may help in the `Build and push` step, but didn't test it...

Reply:
I will implement pushing to DockerHub in another PR.